pkg/test/framework/components/bookinfo/kube.go

```diff
@@ -33,20 +33,24 @@ const (
 	BookinfoDb        bookInfoConfig = "bookinfo-db.yaml"
 )
 
-func deploy(ctx resource.Context, cfg bookInfoConfig) (i deployment.Instance, err error) {
-	ns, err := namespace.Claim(ctx, "default")
+func deploy(ctx resource.Context, cfg Config) (i deployment.Instance, err error) {
+	namespaceName := cfg.Namespace
+	if namespaceName == "" {
+		namespaceName = "default"
+	}
+	ns, err := namespace.Claim(ctx, namespaceName)
```

Comment:
You should use `New` instead of `Claim` to avoid collisions.

Reply:
Using Namespace instance itself.. please review the new logic.